File: eau02.py

""" Créez un programme qui affiche toutes les différentes combinaisons de deux nombre entre 00 et 99 dans l’ordre croissant.


Exemples d’utilisation :
$> python exo.py
00 01, 00 02, 00 03, 00 04, ... , 00 99, 01 02, ... , 97 99, 98 99
$>
"""
# 1: commencer par afficher 2 nombre tel que : 00 01, 00 02 par des condition si 
# 2: afficher 
# 2: ensuite afficher : 00 02  
# 3: 
# 4: 
# 5: 
a=0
b=0
c=0
d=0
for  a in range (0, 10):

    for b in range(0, 10):

        for c in range(0, 10):
                 for d in range(0,10):
            
                    print(a, end="")
                    print(b, end=" ")
                    print(c, end="")
                    print(d, end=", ")


        

File: eau02.js
/* Paramètre à l'envers
Créez un programme qui affiche ses arguments reçus à l’envers.
Exemples d’utilisation :
$> python exo.py “Suis” “Je” “Drôle”
Drôle
Je
Suis
$> python exo.py ha ho
ho
ha
$> python exo.py “Bonjour 36”
Bonjour 36
Afficher error et quitter le programme en cas de problèmes d’arguments.

*/

//Parsing
function getArguments() {
    const arguments = process.argv.slice(2);
    return arguments;
}

//Gestion d'erreur 
function isValidArguments(arguments) {
    if (arguments.length < 2) {
        console.error("ce script nécessite minimum deux arguments");
        return false;
    }
    return true
}



function getReversedStrings(strings) { //la function renverse la chaine de caractère
    const reversedArray = [];
    for (let i = strings.length - 1; i >= 0; i--){
        getReversedStrings.push(strings[i]);
    }
    return reversedArray ;
}

//Résolution
function displayReversedArguments() {

  const arguments = getArguments();

  if (!isValidArguments(arguments)){
    return
  }

  const  reversedArguments = getReversedStrings(arguments) 
  return reversedArguments

}

function displayArray(array) {
    for(const element of array){
        console.log(element)
    }
}

displayArray(getReversedStrings())





